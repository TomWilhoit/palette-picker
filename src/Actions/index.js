export const addProjects = (projects) => ({
  type: 'ADD_PROJECTS',
  projects
})

export const addPalettes = (palettes) => ({
  type: 'ADD_PALETTES',
  palettes
})

export const addCurrentProject = (project) => ({
  type: 'ADD_CURRENT_PROJECT',
  project
})