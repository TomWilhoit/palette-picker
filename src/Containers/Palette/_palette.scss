.palette {
  align-items: center;
  display: flex;
  height: 26%;
  margin: 8px 0 0 4px;
}

.click-container {
  display: flex;
  height: 100%;
  width: 85%;
  &:hover {
    background: $console-hover;
    cursor: pointer;
  }
}

.palette-name {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 45%;
}

.new-palette {
  border: 1px solid black;
  border-radius: 10px;
  height: 100%;
  margin: 0 15px;
  .click-container {
    border-radius: 10px;
    width: 100%;
  }
  .palette-name {
    width: 100%;
  }
}
  
.palette-preview {
  display: grid;
  grid-template-columns: 33% 33% 33%;
  grid-template-rows: 25% 25% 25% 25%;
  grid-template-areas:
    "a b ."
    "a b e"
    "c d e"
    "c d .";
  width: 55%;
}

.delete-palette {
  display: flex;
  justify-content: center;
  width: 15%;
  button {
    @include delete-btn;
    @include small-btn;
  }
}

.color1 {
  grid-area: a;
  margin: 4px 2px 2px 4px;
}

.color2 {
  grid-area: b;
  margin: 4px 2px 2px 4px;
}

.color3 {
  grid-area: c;
  margin: 2px 2px 4px 4px;
}

.color4 {
  grid-area: d;
  margin: 2px 2px 4px 4px;
}

.color5 {
  grid-area: e;
  margin: 3px 4px 3px 2px;
}

.active-palette {
  .click-container {
    background-color: $active; 
  }
}