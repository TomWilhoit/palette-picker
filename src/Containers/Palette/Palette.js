import React, { Component } from "react";
import { connect } from "react-redux";
import { addCurrentPalette, removePalette } from "../../Actions/index";
import { deletePalette } from "../../Utils/API";

export class Palette extends Component {
  constructor(props) {
    super(props);
    this.state = {

    };
  }

  handleClick = async () => {
    await this.props.addCurrentPalette(this.props.id);
    this.props.setColors();
    this.props.showPaletteName();
  };

  erasePalette = (id) => {
    this.props.removePalette(id);
    deletePalette(id);
  };

  handleDelete = (e) => {
    e.preventDefault();
    const id = this.props.id;
    this.erasePalette(id);
  };

  render() {
    const color1 = { backgroundColor: `#${this.props.color1}` }
    const color2 = { backgroundColor: `#${this.props.color2}` }
    const color3 = { backgroundColor: `#${this.props.color3}` }
    const color4 = { backgroundColor: `#${this.props.color4}` }
    const color5 = { backgroundColor: `#${this.props.color5}` }

    return(
      <div className="palette" onClick={() => this.handleClick()}>
        <div className="palette-name">
          <h4>{this.props.name}</h4>
        </div>
        <div className="palette-preview">
          <div className="color-box-preview color1" style={color1}></div>
          <div className="color-box-preview color2" style={color2}></div>
          <div className="color-box-preview color3" style={color3}></div>
          <div className="color-box-preview color4" style={color4}></div>
          <div className="color-box-preview color5" style={color5}></div>
        </div>
        <button onClick={this.handleDelete}>Remove Palette</button>
      </div>
    )
  }
}

export const mapDispatchToProps = dispatch => ({
  addCurrentPalette: palette => dispatch(addCurrentPalette(palette)),
  removePalette: palette => dispatch(removePalette(palette))
})

export default connect(null, mapDispatchToProps)(Palette)