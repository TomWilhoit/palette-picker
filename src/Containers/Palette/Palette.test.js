import React from "react";
import { Palette } from "./Palette";
import { mapStateToProps } from "./Palette";
import { mapDispatchToProps } from "./Palette";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import { updateCurrentPalette } from "../../Actions"
import { removePalette } from "../../Actions"

describe("Palette", () => {
  let wrapper;
  let mockProjects = [{ name: "Tom" }];
  beforeEach(() => {
    wrapper = shallow(<Palette projects={mockProjects} />);
  });

  it("should match the snapshot with all data passed in", () => {
    expect(wrapper).toMatchSnapshot();
  });


  it("should map dispatch to props", () => {
    const mockPalette = {name: "Tommy", projectId: 4}
    const mockDispatch = jest.fn();
    const actionToDispatch = updateCurrentPalette(mockPalette);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.updateCurrentPalette(mockPalette);
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });

  it("should map dispatch to props", () => {
    const mockPalette = {name: "Tommy", projectId: 4}
    const mockDispatch = jest.fn();
    const actionToDispatch = removePalette(mockPalette);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.removePalette(mockPalette);
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});