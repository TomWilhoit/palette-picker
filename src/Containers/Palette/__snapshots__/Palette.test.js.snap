// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Palette should match the snapshot with all data passed in 1`] = `
<div
  className="palette active-palette"
>
  <div
    className="click-container"
    onClick={[Function]}
  >
    <div
      className="palette-name"
    >
      <h4 />
    </div>
    <div
      className="palette-preview"
    />
  </div>
  <button
    onClick={[Function]}
  >
    <i
      className="fas fa-times"
    />
  </button>
</div>
`;
